modules/videoio/src/cap_v4l.cpp

```diff
@@ -1006,8 +1006,8 @@ bool CvCaptureCAM_V4L::open(const char* _deviceName)
     CV_Assert(_deviceName);
     CV_LOG_DEBUG(NULL, "VIDEOIO(V4L2:" << _deviceName << "): opening...");
     FirstCapture = true;
-    width = DEFAULT_V4L_WIDTH;
-    height = DEFAULT_V4L_HEIGHT;
+    width = utils::getConfigurationParameterSizeT("OPENCV_VIDEOIO_V4L_DEFAULT_WIDTH", DEFAULT_V4L_WIDTH);
+    height = utils::getConfigurationParameterSizeT("OPENCV_VIDEOIO_V4L_DEFAULT_HEIGHT", DEFAULT_V4L_HEIGHT);
```

Comment:
What is about `const VideoCaptureParameters& params` approach?

Reply:
Unfortunately it won't work. The VideoCaptureParameters parameter is applied only after the fn_createCaptureCamera_ call. Here is the sequence:
- The `createCapture` is called with the `VideoCaptureParameters`
- The `createCapture` calls the configured `fn_createCaptureCamera_` function of the capture backend (which is `create_V4L_capture_cam in this case`) and after the call is succeeded sets the VideoCaptureParameters. 
The problem is this is the `create_V4L_capture_cam` which is failed (inside the `CvCaptureCAM_V4L::initCapture` function) so we have no chance to apply the VideoCaptureParameters. At least without major refactoring of entire factory code.

Note that the utils::getConfigurationParameter... approach is already used here for the `normalizePropRange` property so the change will not introduce new approaches.